pkg/sentry/kernel/fd_table.go

```diff
@@ -239,6 +247,9 @@ func (f *FDTable) NewFDs(ctx context.Context, fd int32, files []*fs.File, flags
 		for _, i := range fds {
 			f.set(i, nil, FDFlags{}) // Zap entry.
 		}
+		if fd == f.next {
+			f.next = fds[len(fds)-1] + 1
+		}
```

Comment:
Shouldn't this be in the successful case, not the failed one? Also, shouldn't it compare the original value of fd, not the final one? a la

// If we started searching at or before f.next, the next search for an available fd should start after the last fd we return.
if origfd <= f.next {
  f.next = fds[len(fds)-1]+1
} //@nixprime

Reply:
Let me describe my thought:
```c
int fd = open("a");
int fd2 = dup2(fd, 5);
```
Now we hold fd: 0, 1, 2(stdin stdout stderr), 3(file "a"), 5(also file "a").
Before `dup2(fd, 5)`, `next` points to 4. 

In your design, `next` should point to 6, after `dup2(fd, 5)`. 
But in my opioin, `next` points to 4 is correct after `dup2(fd, 5)`.